src/main/java/org/jenkinsci/plugins/envinject/service/EnvInjectEnvVars.java

```diff
@@ -292,4 +313,31 @@ private String removeEscapeDollar(String value) {
         return value;
     }
 
+    public Map<String, String> removeSelfRef(Map<String, String> envVars) {
+        Map<String, String> result = new HashMap<String, String>();
+        for (Map.Entry<String, String> entry : envVars.entrySet()) {
+            if(isSelfRef(entry))
+            {
+                result.put(entry.getKey(), removeSelfRef(entry));
+                logger.info(String.format("Clean self referencing '%s' variable.", entry.getKey()));
+}
+        }
+        return result;
+    }
+    
+    private String removeSelfRef(Map.Entry<String, String> env) {
+        //We replace self variable name unless we are on Windows (Unix only)
+        if ('/' == File.separatorChar) { //unix test
```

Comment:
I doubt it may work in such way, because File.separatorChar will indicate Jenkins master instead of the target node